src/System.IO.FileSystem/tests/Directory/GetFiles_str_str_so.cs

```diff
@@ -8,642 +8,321 @@
-                    Eval(files.Length, list.Count, "Err_969vnk! wrong file count.");
-                    for (int i = 0; i < files.Length; i++)
-                    {
-                        if (Eval(list.Contains(files[i]), "Err_353ygu! unexpected file found: {0}", files[i]))
-                            list.Remove(files[i]);
-                    }
-                    if (!Eval(list.Count == 0, "Err_830vvw! {0} expected files not found.", list.Count))
-                    {
-                        foreach (String fileName in list)
-                            Console.WriteLine(fileName);
-                    }
-                }
+                if (Eval(list.Contains(files[i]), "Err_353ygu! unexpected file found: {0}", files[i]))
+                    list.Remove(files[i]);
             }
-            catch (Exception ex)
+            if (!Eval(list.Count == 0, "Err_830vvw! {0} expected files not found.", list.Count))
             {
-                s_pass = false;
-                Console.WriteLine("Err_983ist! Exception caught in scenario: {0}", ex);
+                foreach (String fileName in list)
+                    Console.WriteLine(fileName);
             }
+        }
+    }
 
-            //Scenario 6: Different local drives, Network drives (UNC and drive letters)
-            /* Scenario disabled when porting because it modifies the filesystem outside of the test's working directory
-            try
-            {
-                string otherDriveInMachine = IOServices.GetNonNtfsDriveOtherThanCurrent();
-                if (otherDriveInMachine != null)
-                {
-                    dirName = ManageFileSystem.GetNonExistingDir(otherDriveInMachine, ManageFileSystem.DirPrefixName);
-                    using (ManageFileSystem fileManager = new ManageFileSystem(dirName, 3, 100))
-                    {
-                        expectedFiles = fileManager.GetAllFiles();
-                        list = new List<String>(expectedFiles);
-                        files = Directory.GetFiles(dirName, "*.*", SearchOption.AllDirectories);
-                        Eval(files.Length, list.Count, "Err_573sdo! wrong file count.");
-                        for (int i = 0; i < files.Length; i++)
-                        {
-                            if (Eval(list.Contains(files[i]), "Err_778kdo! unexpected file found: {0}", files[i]))
-                                list.Remove(files[i]);
-                        }
-                        if (!Eval(list.Count == 0, "Err_954xcx! {0} expected files not found.", list.Count))
-                        {
-                            foreach (String fileName in list)
-                                Console.WriteLine(fileName);
-                        }
-                    }
-                }
+    [Fact]
+    public void Scenario7()
+    {
+        //Scenario 7: dir is readonly
+        String dirName;
+        String[] expectedFiles;
+        String[] files;
+        List<String> list;
```

Comment:
These variables don't all need to be declared at the top of the function (same goes for other tests where this is being done).